### ISSUE TITLE
Some of the _2 tests have some type arguments breaking the corresponding bounds

### ISSUE THREAD

**Issue by stefantsov** [dart-lang/sdk#34613]
I'm seeing those tests while I'm working on adding the checks against the bounds in [CL 57621](https://dart-review.googlesource.com/c/sdk/+/57621).  I believe the tests should be updated.  I'll be marking the tests in status files using the id of this GitHub issue.

One example of such tests is `tests/lib_2/mirrors/generic_f_bounded_mixin_application_test.dart`.  In line 31 there `Collection` is passed as the type argument to `Serializer`, but it does not extend `Serializable<Collection>` as required by the bound.

**Comment by eernstg** [comment]
Agreed that it is a compile-time error in line 31. That situation was a static _warning_ in Dart 1, so the test has probably been ported to Dart 2 too quickly (the only change compared to `lib` is that `@MirrorsUsed(...)` was deleted ;-). I guess the meaningful approach today would be to make `Collection` satisfy the requirement by adding an `implements` clause.

**Comment by stefantsov** [comment]
Ok, I think I have the full list of tests in question:

LanguageFeatures/Instantiate-to-bound/interface/interface_neg_assign_l2_t01
LanguageFeatures/Instantiate-to-bound/interface/interface_neg_assign_l2_t02
LanguageFeatures/Instantiate-to-bound/check_types/typedef_param_t02
LanguageFeatures/Instantiate-to-bound/function/function_named_extends_neg_assign_l1_t10
LanguageFeatures/Instantiate-to-bound/function/function_named_extends_pos_l1_t05
LanguageFeatures/Instantiate-to-bound/function/function_named_extends_pos_l1_t06
LanguageFeatures/Instantiate-to-bound/function/function_optparam_extends_neg_assign_l1_t10
LanguageFeatures/Instantiate-to-bound/function/function_optparam_extends_pos_l1_t05
LanguageFeatures/Instantiate-to-bound/function/function_optparam_extends_pos_l1_t06
LanguageFeatures/Instantiate-to-bound/function/function_param_extends_neg_assign_l1_t10
LanguageFeatures/Instantiate-to-bound/function/function_param_extends_pos_l1_t05
LanguageFeatures/Instantiate-to-bound/function/function_pos_l1_t01
LanguageFeatures/Instantiate-to-bound/function/function_ret_extends_neg_assign_l1_t19
LanguageFeatures/Instantiate-to-bound/interface/interface_neg_assign_l2_t01
LanguageFeatures/Instantiate-to-bound/interface/interface_neg_assign_l2_t02
LanguageFeatures/Instantiate-to-bound/typedef/typedef_named_extends_pos_l2_t01
LanguageFeatures/Instantiate-to-bound/typedef/typedef_optparam_extends_pos_l2_t01
LanguageFeatures/Instantiate-to-bound/typedef/typedef_param_extends_pos_l2_t01
LanguageFeatures/Instantiate-to-bound/typedef/typedef_ret_extends_pos_l2_t01
instantiate_to_bounds_super_bounded_test/54
instantiate_to_bounds_super_bounded_test/55
instantiate_to_bounds_super_bounded_test/56
instantiate_to_bounds_super_bounded_test/57
instantiate_to_bounds_super_bounded_test/58
mirrors/generic_f_bounded_mixin_application_test

**Comment by stefantsov** [comment]
Found some more of such tests:

LanguageFeatures/Instantiate-to-bound/function/function_named_extends_Future_pos_l1_t05
LanguageFeatures/Instantiate-to-bound/function/function_named_extends_Future_pos_l1_t06
LanguageFeatures/Instantiate-to-bound/function/function_optparam_extends_Future_pos_l1_t05
LanguageFeatures/Instantiate-to-bound/function/function_optparam_extends_Future_pos_l1_t06
LanguageFeatures/Instantiate-to-bound/function/function_param_extends_Future_pos_l1_t05
LanguageFeatures/Instantiate-to-bound/function/function_ret_extends_Future_neg_assign_l1_t19
